br/pkg/lightning/restore/table_restore.go

```diff
@@ -886,9 +889,49 @@ func (tr *TableRestore) compareChecksum(remoteChecksum *RemoteChecksum, localChe
 	return nil
 }
 
+func (tr *TableRestore) genAnalyzeSQL(ctx context.Context, g glue.SQLExecutor) string {
+	// TODO: because tidb v5.2 analyze full table OOM easily, we fallback to add `WITH NUM 10000` parameter
+	// if there are more than 600 regions in this table
+	sql := "ANALYZE TABLE "+tr.tableName
+	result, err := g.QueryStringsWithLog(ctx, "SHOW VARIABLES LIKE 'tidb_analyze_version'", "fetch tidb analyze version", tr.logger)
```

Comment:
why not just `select @@tidb_analyze_version` 😕

Reply:
I originally want to be compatible with v4.0, but It's simplier to fallback to 1 on error.